msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -849,8 +815,23 @@ private static ISingleAccountPublicClientApplication createSingleAccountPublicCl
     private static void create(@NonNull final PublicClientApplicationConfiguration config,
                                @Nullable final String clientId,
                                @Nullable final String authority,
+                               @Nullable final String redirectUri,
                                @NonNull final ApplicationCreatedListener listener) {
+        if (clientId != null) {
```

Comment:
why `null` checks here, since we are already validating in public api's above.

Reply:
Some API are not passing this value (i.e. the one that only takes config file).

Reply:
Basically, this function is an 'end point' of all PCA factory API.

Reply:
shouldn't we throw here if it's null then?

Reply:
you mean we will get from config if it's null here ?

Reply:
yes

Reply:
for this function, config is @Nonnull. The caller is responsible for generating the default config file if it's not passed by the app.